age/src/cli_common.rs

```diff
@@ -221,8 +221,16 @@ pub fn read_secret(
     prompt: &str,
     confirm: Option<&str>,
 ) -> pinentry::Result<SecretString> {
-    if let Some(mut input) = PassphraseInput::with_default_binary() {
-        // pinentry binary is available!
+    // Check for the pinentry environment variable. If it's not present try to use the default
+    // binary.
+    let input = if let Ok(pinentry) = std::env::var("RAGE_PINENTRY") {
```

Comment:
Per my comment https://github.com/str4d/rage/issues/280#issuecomment-1587654799, this should not be `rage`-specific:
```suggestion
    let input = if let Ok(pinentry) = std::env::var("PINENTRY_PROGRAM") {
```